### ISSUE TITLE
I think I broke the flow events

### ISSUE THREAD

**Issue by philbooth** [mozilla/fxa-auth-server#2724]
Look at this chart of auth server flow event counts:

https://sql.telemetry.mozilla.org/queries/59968/source#154973

From 27th October, the numbers for all event types emitted by the auth server drops significantly.

Recently I made a number of changes to the metrics context data, deprecating it on some routes and depending on memcached to propagate it for us instead. Memcached has an expiry limit of 2 hours, which might explain some reduction, but I wouldn't expect it to be as great as this.

Related PRs:

* https://github.com/mozilla/fxa-content-server/pull/6639
* https://github.com/mozilla/fxa-auth-server/pull/2666
* https://github.com/mozilla/fxa-auth-server/pull/2677
* https://github.com/mozilla/fxa-auth-server/pull/2682
* https://github.com/mozilla/fxa-auth-server/pull/2682
* https://github.com/mozilla/fxa-auth-server/pull/2683

**Comment by philbooth** [comment]
You can also see it starkly in the chart for DAU:

https://sql.telemetry.mozilla.org/queries/4776#9696

And in the file sizes of our flow data in S3:

```
2018-10-21 00:15:06 4045343199 flow-2018-10-20.csv
2018-10-22 00:15:06 3932935437 flow-2018-10-21.csv
2018-10-23 00:15:07 5227424333 flow-2018-10-22.csv
2018-10-24 00:15:06 6514485310 flow-2018-10-23.csv
2018-10-25 00:15:06 9378969876 flow-2018-10-24.csv
2018-10-26 00:15:06 8769145302 flow-2018-10-25.csv
2018-10-27 00:15:07 10619665645 flow-2018-10-26.csv
2018-10-28 00:15:06 7140798654 flow-2018-10-27.csv
2018-10-29 00:15:06 4813687243 flow-2018-10-28.csv
2018-10-30 00:15:05 6195698918 flow-2018-10-29.csv
2018-10-31 00:15:06 4908660839 flow-2018-10-30.csv
2018-11-01 00:15:09 4040023468 flow-2018-10-31.csv
2018-11-02 00:15:15 3876141861 flow-2018-11-01.csv
2018-11-03 00:15:09 3497174537 flow-2018-11-02.csv
2018-11-04 00:15:10 2965643096 flow-2018-11-03.csv
2018-11-05 00:15:04  441237869 flow-2018-11-04.csv
2018-11-06 00:15:10   70018037 flow-2018-11-05.csv
2018-11-07 00:15:07   72965831 flow-2018-11-06.csv
```

**Comment by irrationalagent** [comment]
Part of the weirdness here is that there is also another major drop in both content and auth server event volume that happens about a week later between nov 3rd to 4th (very close to train 124 but appears to precede it by about a day), unclear at this point if its due to a separate issue:

https://sql.telemetry.mozilla.org/queries/59966/source#154969

Furthermore, there does not seem to be a real change in amplitude volume, which should probably tell us something since so many amplitude events are based off the flow events:

https://analytics.amplitude.com/mozilla-corp/chart/gzbnuzm

**Comment by philbooth** [comment]
This is a genuine mystery. I just want to record some things I've looked at/thought about before I go off for a walk in search of inspiration.

* In local dev, all flow events are being emitted as expected when signing in to a web session.
* In local dev, all flow events are being emitted as expected when signing in to Sync.
* In local dev, all flow events are being emitted as expected when resetting a password.
* In local dev, I haven't tried about:welcome or new tab yet. I will do once vladikoff/fxa-dev-launcher#4 lands.
* I wonder whether something could have happened with memcached in prod, so flow ids are getting evicted very quickly before they can be used?
* I wonder whether something could have happened with the `flow_id`/`flow_begin_time` query params, so they're corrupted and fail validation when we try to emit events?

**Comment by philbooth** [comment]
> I haven't tried about:welcome or new tab yet.

Fwiw this works fine locally too, all flow events are emitted as expected.

**Comment by philbooth** [comment]
> I wonder whether something could have happened with memcached in prod, so flow ids are getting evicted very quickly before they can be used?

I asked @jbuck and @jrgm about this, there were no evictions around the 27th October. There was increased traffic from the Firefox release, but nothing that looks like it could explain this issue.

**Comment by philbooth** [comment]
Just realised I should have asked our ops team another question last night, which was whether the flow event volumes in our raw logs took a hit, either on 27th October or 4th November.

Because if they didn't it means they must be going missing in the Heka part of the import pipeline (I'm presuming it still uses Heka because we never heard to the contrary, but I haven't actually asked anyone). And if they did take a hit, it's still a mystery but at least it further narrows down the possibilities.

**Comment by jbuck** [comment]
On Nov 3rd the Heka aggregator stopped accepting logs. whd restarted it and we've start processing the backlog

**Comment by irrationalagent** [comment]
Content erver event volume has dropped to 0 since the 7th.

**Comment by philbooth** [comment]
Event volumes appear to pick back up from Saturday 10th November, optimistically closing this.
